File: ios/Source/Extensions/StoreKeys+WalletLink.swift
// Copyright (c) 2018-2019 Coinbase, Inc. <https://coinbase.com/>
// Licensed under the Apache License, version 2.0

import CBStore

extension StoreKeys {
    /// Store key to keeping track of WalletLink sessions
    static let sessions = KeychainStoreKey<SessionList>(
        "walletlink_session_list",
        accessible: .afterFirstUnlockThisDeviceOnly
    )
}

File: android/walletlink/src/main/java/com/coinbase/walletlink/extensions/StoreKeys+WalletLink.kt
// Copyright (c) 2018-2019 Coinbase, Inc. <https://coinbase.com/>
// Licensed under the Apache License, version 2.0

package com.coinbase.walletlink.extensions

import com.coinbase.wallet.store.models.EncryptedSharedPrefsStoreKey
import com.coinbase.wallet.store.models.StoreKeys
import com.coinbase.walletlink.models.Session

/**
 * Store key to keeping track of WalletLink sessions
 */
internal val StoreKeys.sessions: EncryptedSharedPrefsStoreKey<Array<Session>> by lazy {
    EncryptedSharedPrefsStoreKey(id = "walletlink_session_list", clazz = Array<Session>::class.java)
}
